presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java

```diff
@@ -182,7 +182,7 @@ public static long localTimestamp(SqlFunctionProperties properties)
     @SqlType(StandardTypes.TIMESTAMP)
     public static long fromUnixTime(@SqlType(StandardTypes.DOUBLE) double unixTime)
     {
-        return Math.round(unixTime * 1000);
+        return Math.round(Math.floor(unixTime) * 1000 + Math.round((unixTime - Math.floor(unixTime)) * 1000));
```

Comment:
This is a tricky change. Future readers won't know why this is done in this particular way and 'git log' won't help much because commit message just says 'loses precision in some corner cases' without providing any details.

Would you add a comment here to explain what's going on and why the computation is done this way and also update commit message to explain clearly the problem and solution?

Reply:
@hainenber 
We could reuse some sentences from the issue and put them in comment.

Reply:
I've amended the change with more details in the form of comments and commit message.